From 484288d09a0e04fa30d24c131f2ab87dc2bece99 Mon Sep 17 00:00:00 2001
From: Dmitry Vyukov <dvyukov@google.com>
Date: Wed, 12 May 2021 09:07:00 +0200
Subject: [PATCH] scudo: fix CheckFailed-related build breakage

I was running:

$ ninja check-sanitizer check-msan check-asan \
  check-tsan check-lsan check-ubsan check-cfi \
  check-profile check-memprof check-xray check-hwasan

but missed check-scudo...

Differential Revision: https://reviews.llvm.org/D102314
---
 compiler-rt/lib/scudo/scudo_termination.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler-rt/lib/scudo/scudo_termination.cpp b/compiler-rt/lib/scudo/scudo_termination.cpp
index 6c7c0abc6d36..5f1337efaca0 100644
--- a/compiler-rt/lib/scudo/scudo_termination.cpp
+++ b/compiler-rt/lib/scudo/scudo_termination.cpp
@@ -30,7 +30,7 @@ void NORETURN Die() {
   internal__exit(common_flags()->exitcode);
 }
 
-void SetCheckFailedCallback(CheckFailedCallbackType callback) {}
+void SetCheckUnwindCallback(void (*callback)()) {}
 
 void NORETURN CheckFailed(const char *File, int Line, const char *Condition,
                           u64 Value1, u64 Value2) {
-- 
2.39.3

